### PR TITLE
Rename fb⇝Meta in fbOnly block, try to fix #1129

### LINKED ISSUE #1129: Remove leading 'fbOnly' sections from the DOM if non-FB viewer
This is most important for API reference pages.

Right now, we hide them with CSS. The problem with this is it shows up in Google results (e.g. search for 'vec map hhvm', and 'Facebook Engineer? This is available as Vec\map in the www repository. ' is just a waste of space there and makes google less useful.

This is not about confidentially etc, as the site (and all of its' inputs) are open source and made on public systems. Just about making the text that google shows more useful.

This isn't quite trivial as we currently pregenerate the entire `.html` file. There's a few options:
- filter them out of the HTML at render time (brittle, slow, but straightforward)
- stop putting them in the HTML. Inject at render time (only works for leading fbonly messages, unless very complicated)
- render separate _fb.html files. Show either foo.html or foo_fb.html depending on ASN

For all of these, it would probably be worth adding a markdown extension to explicitly recognize fbonly, instead of using `<div class='fbOnly'>`, so they can be more-reliably matched.

### PR DESCRIPTION
**Pull request by walaura**
This seems like it should work. Fixes #1129 (maybe)

Google's docs on `data-nosnippet`:
https://developers.google.com/search/docs/advanced/robots/robots_meta_tag

Happy to split the commits! i was going for a stacked diff sorta thing here but it's been a while since i used GH ><

### CODE REVIEW

File: src/markdown-extensions/YamlFrontMatterBlock.php [hunk 1]
```diff
@@ -127,13 +127,13 @@ private static function getFacebookMessages(
...
-        '<div class="message api fbOnly">'.
-        "**Facebook Engineer?**\n\n".
+        '<div class="message api fbOnly" data-nosnippet>'.
+        "**Meta Engineer?**\n\n".
         '<p>'.
         $message.
         "</p>\n".
         '<!-- '.
-        'Not a Facebook engineer... yet? https://www.facebook.com/careers/'.
+        'Not a Meta engineer... yet? https://www.facebook.com/careers/'.
```
Comment (demosdemon) [inline src/markdown-extensions/YamlFrontMatterBlock.php:136]: https://metacareers.com/

Comment (walaura) [reply src/markdown-extensions/YamlFrontMatterBlock.php]: done!

**Comment by walaura** [comment]
hey thanks! hopefully we get a clean merge because i got 0 tooling for this so i used codespaces! best of luck with the fix, hope it works 🥹

**Comment by AndrewDiMola** [comment]
Merged in https://github.com/hhvm/user-documentation/pull/1263

**Pull request closed without merging**